iroha_error/src/lib.rs

```diff
@@ -1,196 +1,145 @@
-//!
-//! Example:
-//! ```rust
-//! # pub trait Deserialize {}
-//! #
-//! # mod serde_json {
-//! #     use super::Deserialize;
-//! #     use std::io;
-//! #
-//! #     pub fn from_str<T: Deserialize>(json: &str) -> io::Result<T> {
-//! #         unimplemented!()
-//! #     }
-//! # }
-//! #
-//! # struct ClusterMap;
-//! #
-//! # impl Deserialize for ClusterMap {}
-//! #
-//! use iroha_error::Result;
-//!
-//! fn get_cluster_info() -> Result<ClusterMap> {
-//!     let config = std::fs::read_to_string("cluster.json")?;
-//!     let map: ClusterMap = serde_json::from_str(&config)?;
-//!     Ok(map)
-//! }
-//! # fn main() {}
-//! ```
-//!
-//! Wrap a lower level errors with some context:
-//! ```rust
-//! # struct It;
-//! #
-//! # impl It {
-//! #     fn detach(&self) -> Result<()> {
-//! #         unimplemented!()
-//! #     }
-//! # }
-//! #
-//! use iroha_error::{WrapErr, Result};
-//!
-//! fn main() -> Result<()> {
-//!     # return Ok(());
-//!     #
-//!     # const _: &str = stringify! {
-//!     ...
-//!     # };
-//!     #
-//!     # let it = It;
-//!     # let path = "./path/to/instrs.json";
-//!     #
-//!     it.detach().wrap_err("Failed to detach the important thing")?;
-//!
-//!     let content = std::fs::read(path)
-//!         .wrap_err_with(|| format!("Failed to read instrs from {}", path))?;
-//!     #
-//!     # const _: &str = stringify! {
-//!     ...
-//!     # };
-//!     #
-//!     # Ok(())
-//! }
-//! ```
-
-#![allow(clippy::module_name_repetitions)]
-
 use std::{
-    convert::{AsRef, From},
+    env,
     error::Error as StdError,
-    fmt::{self, Debug, Display},
-    ops::{Deref, DerefMut},
-    result::Result as StdResult,
+    fmt::{self, Display, Formatter},
+    panic,
+    panic::PanicInfo,
+    path::Path,
+    ptr,
+    sync::atomic::{AtomicBool, Ordering},
 };
 
-pub use message_error::MessageError;
-pub use reporter::{install as install_panic_reporter, Reporter};
-pub use wrap_err::{WrapErr, WrappedError};
-
-pub mod reporter;
-
-/// Module with derive macroses
-pub mod derive {
-    /// Derive macro for enums which implements source and display for it.
-    ///
-    /// ```rust
-    /// #[derive(iroha_error::derive::Error, Debug)]
-    /// enum Error {
-    ///     #[error("Failed because of reason a")]
-    ///     ErrorA,
-    ///     // #[source] tells macro to use io::Error as source for this variant
-    ///     #[error("Failed during reading file")]
-    ///     IOError(#[source] std::io::Error),
-    /// }
-    /// ```
-    pub use iroha_error_macro::Error;
-}
-
-mod message_error;
-mod wrap_err;
-
-/// Error type similar to `anyhow::Error`, `eyre::Reporter`.
-pub struct Error {
-    inner: Box<dyn StdError + Send + Sync + 'static>,
-}
-
-/// Result type which uses Error
-pub type Result<T, E = Error> = StdResult<T, E>;
-
-impl Display for Error {
-    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-        Display::fmt(&self.inner, formatter)
+use owo_colors::OwoColorize;
```

Comment:
Could you remove whole crate `iroha_error`? This reporter `Backtrace` is no use now.

Reply:
When discussing with me I think @KalitaAlexey mentioned that we should leave iroha_error as it contains a panic reporter that is used in other crates

Reply:
Yeah that does make sense. But this strange though. We can have better reporter:
https://docs.rs/color-eyre/0.5.11/color_eyre/index.html

Reply:
Sure, let's use this

Reply:
Is the default theme of color-eyre good enough?

Reply:
Yep

Reply:
I think ok